src/api/Feed.js

```diff
@@ -0,0 +1,1022 @@
+        successCallback: Function
+    ): void => {
+        const cachedItems = this.getCachedItems(this.id);
+        if (cachedItems) {
+            // $FlowFixMe
+            const task: ?Task = cachedItems.items.find((item) => item.type === 'task' && item.id === taskId);
+            if (task) {
+                const { entries, total_count } = task.task_assignment_collection;
+                const assignments = entries.map((item: TaskAssignment) => {
+                    if (item.id === updatedAssignment.id) {
+                        return {
+                            ...item,
+                            ...updatedAssignment,
+                            resolution_state: updatedAssignment.message.toLowerCase()
+                        };
+                    }
+
+                    return item;
+                });
+
+                this.updateFeedItem(
+                    {
+                        task_assignment_collection: {
+                            entries: assignments,
+                            total_count
+                        },
+                        isPending: false
+                    },
+                    taskId
+                );
+                successCallback(updatedAssignment);
+            }
+        }
+    };
+
+    /**
+     * Updates a task
+     *
+     * @param {BoxItem} file - The file to which the task is assigned
+     * @param {string} taskId - The task's id
+     * @param {Array} message - The task's text
+     * @param {Function} successCallback - the function which will be called on success
+     * @param {Function} errorCallback - the function which will be called on error
+     * @param {string} dueAt - The optional date the task is due
+     * @return {void}
+     */
+    updateTask = (
+        file: BoxItem,
+        taskId: string,
+        message: string,
+        successCallback: (task: Task) => void = noop,
+        errorCallback: (e: $AxiosXHR<any>) => void = noop,
+        dueAt?: string
+    ) => {
+        if (!file.id) {
+            throw getBadItemError();
+        }
+
+        this.id = file.id;
+        this.updateFeedItem({ isPending: true }, taskId);
+        this.tasksAPI = new TasksAPI(this.options);
+        this.tasksAPI.updateTask({
+            file,
+            taskId,
+            message,
+            dueAt,
+            successCallback: (task: Task) => {
+                this.updateTaskSuccessCallback(task, successCallback);
+            },
+            errorCallback: (e: $AxiosXHR<any>) => {
+                this.updateTaskErrorCallback(e, errorCallback);
+            }
+        });
+    };
+
+    /**
+     * Update task error callback
+     *
+     * @param {Object} e - Axios error
+     * @param {Function} errorCallback - The error callback
+     * @return {void}
+     */
+    updateTaskErrorCallback = (e: $AxiosXHR<any>, errorCallback: Function) => {
+        this.errorCallback(e);
+        if (!this.isDestroyed()) {
+            errorCallback(e);
```

Comment:
Regardless of this being destroyed, shouldn't a passed in error callback always fire when an error occurred? How will the parent of this know what to do?

Reply:
If the API is destroyed, the parent shouldnt care about the response anymore